src/pybamm/models/submodels/porosity/reaction_driven_porosity.py

```diff
@@ -37,7 +38,9 @@ def get_coupled_variables(self, variables):
                 L_pl_k = variables[f"{Domain} lithium plating thickness [m]"]
                 L_dead_k = variables[f"{Domain} dead lithium thickness [m]"]
                 L_sei_cr_k = variables[f"{Domain} total SEI on cracks thickness [m]"]
-                roughness_k = variables[f"{Domain} electrode roughness ratio"]
+                roughness_k = variables[
+                    f"{Domain} electrode {self.phase_name}roughness ratio"
```

Comment:
The issue with the `phase_name` is that this class does not take a `phase` as argument (see for example how `loss_active_material.py` does it. However, I don't understand how this model works: you have different values for different phases for the roughness, but the SEI on cracks thickness is only one. Shouldn't you just have one roughness ratio overall?

Reply:
@DrSOKane could you please advise on how I can go about with introducing phases in the reaction-driven-porosity model? Most of the remaining errors are stemming from this file. Thank you.

Reply:
Hi @mohammedasher, you need to move the roughness ratio inside the loop `for phase in phases:`. Something like this should work:

`roughness_k = variables[
    f{Domain} total {phase_name}SEI on cracks thickness [m]"
]`

Then, delete the reference to `roughness_k` outside the loop (line 36 in my version). `phase_name` is defined at the start of the loop `for phase in phases:` so that gets around the problems Ferran rightly pointed out.

Reply:
Hi @DrSOKane....I will define a new `for phase in phases:` loop; apart from the `roughness_k` variable, `L_sei_k = variables[f"{Domain} total SEI thickness [m]"] ` also runs into error when the domain is negative (negative electrode). I tried `L_sei_k = variables[f"{Phase}: {Domain} total SEI thickness [m]"]', but it did not help.

Reply:
@mohammedasher the code you need is [here](https://github.com/DrSOKane/PyBaMM/blob/composite-porosity/src/pybamm/models/submodels/porosity/reaction_driven_porosity.py). Uncomment lines 63-65 and deleted line 36, and it should work.